spec/requests/api/v2/providers/suggest_provider_spec.rb

```diff
@@ -0,0 +1,59 @@
+require 'rails_helper'
+
+describe 'GET /suggest' do
+  let(:jsonapi_renderer) { JSONAPI::Serializable::Renderer.new }
+  let(:organisation) { create(:organisation) }
+  let(:provider1) { create(:provider, organisations: [organisation]) }
+  let(:provider2) { create(:provider, organisations: [organisation]) }
+  let(:next_recruitment_cycle) { find_or_create(:recruitment_cycle, :next) }
+  let(:provider3) { create(:provider, provider_name: provider2.provider_name, organisations: [organisation], recruitment_cycle: next_recruitment_cycle) }
+  let(:user) { create :user, organisations: [organisation] }
+  let(:payload) { { email: user.email } }
+  let(:token) { build_jwt :apiv2, payload: payload }
+  let(:credentials) do
+    ActionController::HttpAuthentication::Token.encode_credentials(token)
+  end
+
+  before do
+    provider1
+    provider2
+    provider3
+  end
+
+  it 'searches for a particular provider for the user for the current recruitment cycle' do
+    get "/api/v2/providers/suggest?query=#{provider2.provider_name}",
+        headers: { 'HTTP_AUTHORIZATION' => credentials }
+
+    expect(JSON.parse(response.body)['data']).to match_array([
```

Comment:
also like to see

it 'searches for a non existing provider for the user for the current recruitment cycle' do
    get "/api/v2/providers/suggest?query=#{non existing}",